src/result.cpp

```diff
@@ -116,30 +113,26 @@ read_embedded_file_entry(CacheEntryReader& reader,
                          const ResultFileMap* result_file_map,
                          FILE* dump_stream)
 {
-  uint8_t suffix_len;
-  reader.read(suffix_len);
-
-  char suffix[256];
-  reader.read(suffix, suffix_len);
+  FileType type;
+  reader.read(*reinterpret_cast<UnderlyingFileTypeInt*>(&type));
```

Comment:
I find these casts a bit ugly. How about working with an `uint8_t` instead and casting when searching instead, i.e. something like

```c++
uint8_t type;
reader.read(type);
// ...
const auto it = result_file_map->find(FileType(type));
```

(or use `UnderlyingFileTypeInt`)? And similar in other places. What do you think?